timemachine/cpp/src/custom_nonbonded_gpu.cu

```diff
@@ -130,21 +132,42 @@ void Electrostatics<RealType>::derivatives_device(
     dim3 dimGrid(n_blocks, dim_y, C); // x, y, z dims
 
     // auto start = std::chrono::high_resolution_clock::now();
-    k_electrostatics<<<dimGrid, dimBlock>>>(
-        N,
-        d_coords,
-        d_params,
-        d_scale_matrix_,
-        d_param_idxs_,
-        d_E,
-        d_dE_dx,
-        d_d2E_dx2,
-        // parameter derivatives
-        num_dp,
-        d_param_gather_idxs,
-        d_dE_dp,
-        d_d2E_dxdp
-    );
+    if(num_dims == 3) {
```

Comment:
@bp-kelley if only we could JIT compile templates at runtime. I don't have any better ways to do this than to specialize apriori, any better ideas?